src/mpi/comm/comm_split_type.c

```diff
@@ -579,14 +714,8 @@ int MPIR_Comm_split_type_neighborhood(MPIR_Comm * comm_ptr, int split_type, int
         *newcomm_ptr = dummycomm_ptr;
 #endif
     } else {
-        /* In the mean time, the user passed in
-         * COMM_TYPE_NEIGHBORHOOD but did not give us an info we
-         * know how to work with.  Throw up our hands and treat it
-         * like UNDEFINED.  This will result in MPI_COMM_NULL
-         * being returned to the user. */
-        *newcomm_ptr = NULL;
+        MPIR_Comm_split_type_network_topo(comm_ptr, key, info_ptr, newcomm_ptr);
```

Comment:
This doesn't sound right.  If the user didn't give a valid info key, we call `MPIR_Comm_split_type_network_topo`?